single-line-diagram-core/src/test/java/com/powsybl/sld/TestSVGWriter.java

```diff
@@ -0,0 +1,408 @@
+        substG.addEdge(vl1Trf2One, vl2Trf2One);
+        substG.getEdges().get(substG.getEdges().size() - 1).setSnakeLine(Arrays.asList(360., 80., 360., 50., 680., 50., 680., 80.));
+        substG.addEdge(vl1Trf2Two, vl3Trf2One);
+        substG.getEdges().get(substG.getEdges().size() - 1).setSnakeLine(Arrays.asList(440., 80., 440., 30., 960., 30., 960., 80.));
+        substG.addEdge(vl2Trf2Two, vl3Trf2Two);
+        substG.getEdges().get(substG.getEdges().size() - 1).setSnakeLine(Arrays.asList(740., 80., 740., 10., 1040., 10., 1040., 80.));
+    }
+
+    @Test
+    public void test() {
+        // initValueProvider example for the test :
+        //
+        DiagramInitialValueProvider initValueProvider = new DiagramInitialValueProvider() {
+            @Override
+            public InitialValue getInitialValue(Node node) {
+                InitialValue initialValue;
+                if (node.getType() == Node.NodeType.BUS) {
+                    initialValue = new InitialValue(null, null, node.getLabel(), null, null, null);
+                } else {
+                    initialValue = new InitialValue(Direction.UP, Direction.DOWN, "10", "20", null, null);
+                }
+                return initialValue;
+            }
+
+            @Override
+            public List<String> getNodeLabelValue(Node node) {
+                List<String> res = new ArrayList<>();
+                if (node instanceof FeederNode || node instanceof BusNode) {
+                    res.add(node.getLabel());
+                }
+                return res;
+            }
+        };
+
+        // Layout parameters :
+        //
+        LayoutParameters layoutParameters = new LayoutParameters()
+                .setTranslateX(20)
+                .setTranslateY(50)
+                .setInitialXBus(0)
+                .setInitialYBus(260)
+                .setVerticalSpaceBus(25)
+                .setHorizontalBusPadding(20)
+                .setCellWidth(80)
+                .setExternCellHeight(250)
+                .setInternCellHeight(40)
+                .setStackHeight(30)
+                .setShowGrid(true)
+                .setShowInternalNodes(false)
+                .setScaleFactor(1)
+                .setHorizontalSubstationPadding(50)
+                .setVerticalSubstationPadding(50)
+                .setDrawStraightWires(false)
+                .setHorizontalSnakeLinePadding(30)
+                .setVerticalSnakeLinePadding(30)
+                .setShowInductorFor3WT(true);
+
+        Map<String, Graph> mapGr = new HashMap<>();
+        mapGr.put("vl1.svg", g1);
+        mapGr.put("vl2.svg", g2);
+        mapGr.put("vl3.svg", g3);
+
+        for (String filename : mapGr.keySet()) {
+            // SVG file generation first voltage level and comparison to reference :
+            //
+            Path pathSVG = Paths.get(System.getProperty("user.home"), filename);
```

Comment:
You should never ever rely on local file system for unit testing. Use either in memory String comparison like in compareSvg(...)

Reply:
Done